pkg/common/commands/resetoffset/refmappers/subscription.go

```diff
@@ -0,0 +1,155 @@
+// the Context having injected informers (SubscriptionInformer).
+func (f *SubscriptionRefMapperFactory) Create(ctx context.Context) ResetOffsetRefMapper {
+	return NewSubscriptionRefMapper(ctx, f.TopicNameMapper, f.GroupIdMapper)
+}
+
+//
+// SubscriptionRefMapper
+//
+
+// SubscriptionTopicNameMapper defines a function signature for mapping a Subscription to a Kafka Topic name.
+type SubscriptionTopicNameMapper func(*messagingv1.Subscription) (string, error)
+
+// SubscriptionConsumerGroupIdMapper defines a function signature for mapping a Subscription to a Kafka ConsumerGroup ID.
+type SubscriptionConsumerGroupIdMapper func(*messagingv1.Subscription) (string, error)
+
+// Verify The Subscription ResetOffsetRefMapper Implements The Interface
+var _ ResetOffsetRefMapper = &SubscriptionRefMapper{}
+
+// SubscriptionRefMapper implements the ResetOffsetRefMapper for Knative Subscriptions
+type SubscriptionRefMapper struct {
+	logger             *zap.Logger
+	subscriptionLister messaginglisters.SubscriptionLister
+	topicNameMapper    SubscriptionTopicNameMapper
+	groupIdMapper      SubscriptionConsumerGroupIdMapper
+}
+
+// NewSubscriptionRefMapper returns an initialized ResetOffsetSubscriptionRefMapper
+func NewSubscriptionRefMapper(ctx context.Context,
+	topicNameMapper SubscriptionTopicNameMapper,
+	groupIdMapper SubscriptionConsumerGroupIdMapper) *SubscriptionRefMapper {
+
+	// Get The Logger From Context
+	logger := logging.FromContext(ctx).Desugar()
+
+	// Get The Subscription Informer From Context (Context Must Have Injected Informers From SharedMain())
+	subscriptionInformer := subscriptioninformers.Get(ctx)
+
+	// Return An Initialized ResetOffsetSubscriptionRefMapper
+	return &SubscriptionRefMapper{
+		logger:             logger,
+		subscriptionLister: subscriptionInformer.Lister(),
+		topicNameMapper:    topicNameMapper,
+		groupIdMapper:      groupIdMapper,
+	}
+}
+
+// MapRef implements the ResetOffsetRefMapper interface for Subscription references. It will return an
+// error in all cases other than successfully mapping the ResetOffset.Spec.Ref to a Kafka Topic / Group.
+func (m *SubscriptionRefMapper) MapRef(resetOffset *kafkav1alpha1.ResetOffset) (string, string, error) {
+
+	// Validate The ResetOffset
+	if resetOffset == nil {
+		m.logger.Warn("Received nil ResetOffset argument")
+		return "", "", fmt.Errorf("unable to map nil ResetOffset")
+	}
+
+	// Get The ResetOffset Ref From Spec & Enhance Logger
+	ref := resetOffset.Spec.Ref
+	logger := m.logger.With(zap.Any("Ref", ref))
+
+	// Validate The Reference
+	if !strings.HasPrefix(ref.APIVersion, messaging.GroupName) ||
+		ref.Kind != "Subscription" ||
+		len(ref.Name) <= 0 {
+		m.logger.Warn("Received ResetOffset with invalid Subscription reference")
+		return "", "", fmt.Errorf("received ResetOffset with invalid Subscription reference: %v", ref)
+	}
+
+	// Default Optional Ref.Namespace If Not Provided
+	refNamespace := ref.Namespace
+	if len(refNamespace) <= 0 {
```

Comment:
`refNamespace == ""` IMO is nicer to read, but you don't need to change it if you don't like it :)

Reply:
Yeah, not sure how I got into the habit of that - some misplaced blog early in my go development.  You are right the "" check is easier to read and they are essentially equivalent - I will change it  (and my habitual use of len() as well ; ). thx!